src/analysis/processing/qgsalgorithmpointstopaths.cpp

```diff
@@ -0,0 +1,334 @@
+    email                : uclaros at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmpointstopaths.h"
+#include "qgsvectorlayer.h"
+
+///@cond PRIVATE
+
+QString QgsPointsToPathsAlgorithm::name() const
+{
+  return QStringLiteral( "pointstopath" );
+}
+
+QString QgsPointsToPathsAlgorithm::displayName() const
+{
+  return QObject::tr( "Points to path" );
+}
+
+QString QgsPointsToPathsAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm takes a point layer and connects its features creating a new line layer.\n\n"
+                      "An attribute or expression may be specified to define the order the points should be connected. "
+                      "If no order expression is specified, the fid is used.\n\n"
+                      "A natural sort can be used when sorting by a string attribute "
+                      "or expression (ie. place 'a9' before 'a10').\n\n"
+                      "An attribute or expression can be selected to group points having the same value into the same resulting line." );
+}
+
+QStringList QgsPointsToPathsAlgorithm::tags() const
+{
+  return QObject::tr( "create,lines,points,connect,convert,join" ).split( ',' );
+}
+
+QString QgsPointsToPathsAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsPointsToPathsAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+void QgsPointsToPathsAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ),
+                QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "CLOSE_PATH" ),
+                QObject::tr( "Create closed paths" ), false, true ) );
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "ORDER_EXPRESSION" ),
+                QObject::tr( "Order expression" ), QVariant(), QStringLiteral( "INPUT" ), true ) );
+  addParameter( new QgsProcessingParameterBoolean( QStringLiteral( "NATURAL_SORT" ),
+                QObject::tr( "Sort text containing numbers naturally" ), false, true ) );
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "GROUP_EXPRESSION" ),
```

Comment:
The existing algorithm has "GROUP_FIELD" parameter which also needs to be retained for compatibility

Reply:
Seems like I subconsciously skipped this, since I didn't like the idea of the complete port!! :laughing: